modules/reindex/src/test/java/org/elasticsearch/reindex/ReindexScriptTests.java

```diff
@@ -34,8 +34,8 @@ public void testSetIndex() throws Exception {
     public void testSettingIndexToNullIsError() throws Exception {
         try {
             applyScript((Map<String, Object> ctx) -> ctx.put("_index", null));
-        } catch (NullPointerException e) {
-            assertThat(e.getMessage(), containsString("Can't reindex without a destination index!"));
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), containsString("_index cannot be null"));
```

Comment:
Yeah.  Unfortunately, because it's enforced in `Metadata` based on the `FieldProperty` along with all other Metadata constraints, it's missing some context.

However, it does still indicate the problem.